File: packages/silo-client/src/scope/rename-options.ts
import type { RequestOptions } from "../request-options.js";

/**
 * `rename()`'s options across projects, environments and collections.
 * `expectedId` binds the call to the record a caller already read — from a
 * dry run's report, from `list()`, or supplied directly.
 */
export interface RenameOptions extends RequestOptions {
  expectedId?: string;
  dryRun?: boolean;
}

File: packages/silo-client-dart/lib/src/scope/rename_options.dart
import '../request_options.dart';

/// [expectedId] binds the rename to the record a caller already read.
final class RenameOptions extends RequestOptions {
  const RenameOptions({this.expectedId, this.dryRun = false, super.timeout, super.cancellation});

  final String? expectedId;
  final bool dryRun;
}
